lib/charms/opensearch/v0/opensearch_backups.py

```diff
@@ -424,7 +483,9 @@ def _on_s3_credentials_changed(self, event: EventBase) -> None:
 
         try:
             plugin = self.charm.plugin_manager.get_plugin(OpenSearchBackupPlugin)
-            self.charm.plugin_manager._configure_if_needed(plugin)
+            if self.charm.plugin_manager.status(plugin) == PluginState.ENABLED:
+                self.charm.plugin_manager.disable(plugin)
```

Comment:
Could you explain why we're disabling ?

Reply:
Because it contains configuration from the other plugin, mainly different keys in `OpenSearchKeystore`.

Reply:
the other plugin == S3 plugin?